src/network/dn_comp.c

```diff
@@ -54,24 +89,36 @@ static int match(int *offset, const unsigned char *base, const unsigned char *dn
 	}
 }
 
-int dn_comp(const char *src, unsigned char *dst, int space, unsigned char **dnptrs, unsigned char **lastdnptr)
+// dn_comp compresses the domain name src and stores it in the buffer dst of length space. The
+// compresssion uses an array of pointers dnptrs to previously compressed names in the current
+// message. The first pointer points to the beggining of the message and the list ends with NULL.
+// The limit of the array is specified with lastdnptr. If dnptrs is NULL, domain names are not
+// compressed. If lastdnptr is NULL, the list of labels is not updated.
+int dn_comp(const char *src : itype(_Nt_array_ptr<const char>),
+	unsigned char *dst : count(space - 1) itype(_Nt_array_ptr<unsigned char>),
```

Comment:
The nt_array_ptr `dst` has `count(space - 1)` because its prefix has at most length `space - 1`.